src/test/com/Elessar/app/client/MyClientIntegTest.java

```diff
@@ -0,0 +1,233 @@
+            assertEquals(true, registerResp1.getSuccess());
+
+            RegistrationResponse registerResp2 = clientA.register("Shuai Hu", "4238902", "shuaih@yahoo.com", "62357492");
+            assertEquals(false, registerResp2.getSuccess());
+
+            RegistrationResponse registerResp3 = clientA.register("Zhi Xu", "213252", "zhix@126.com", "62258237");
+            assertEquals(true, registerResp3.getSuccess());
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                assertEquals(true, cursor.hasNext());
+            }
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Zhi Xu")).iterator()) {
+                assertEquals(true, cursor.hasNext());
+            }
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            assert(false);
+        }
+    }
+
+
+    private void logOnTest() {
+        try {
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(false, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+            LogonResponse logonResp1 = clientA.logOn("Ertai Cai", "2222123", clientA_Port);
+            Assert.assertEquals(false, logonResp1.getSuccess());
+
+            LogonResponse logonResp2 = clientA.logOn("Shuai Hu", "4238903", clientA_Port);
+            Assert.assertEquals(false, logonResp2.getSuccess());
+
+            LogonResponse logonResp3 = clientA.logOn("Shuai Hu", "4238902", clientA_Port);
+            Assert.assertEquals(true, logonResp3.getSuccess());
+
+            LogonResponse logonResp4 = clientA.logOn("Shuai Hu", "4238902", clientA_Port);
+            Assert.assertEquals(true, logonResp4.getSuccess());
+
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(true, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            assert(false);
+        }
+    }
+
+
+    private void logOffTest() {
+        try {
+            try (MongoCursor<Document> cursor = mongoDB.getCollection(MyDatabase.USERS).find(eq(User.NAME, "Shuai Hu")).iterator()) {
+                if (cursor.hasNext()) {
+                    assertEquals(true, cursor.next().getBoolean(User.ONLINE));
+                }
+            }
+
+            LogoffResponse logoffResp1 = clientA.logOff("Ertai Cai");
+            Assert.assertEquals(false, logoffResp1.getSuccess());
+
+            LogoffResponse logoffResp2 = clientA.logOff("Zhi Xu");
```

Comment:
Better add a comment here and also for logoffResp3 mentioning what we are testing here: e.g. one is when user registered but offline and the other is user registered and online.